src/main/java/org/terasology/moduletestingenvironment/TestEventReceiver.java

```diff
@@ -0,0 +1,125 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.moduletestingenvironment;
+
+import org.terasology.context.Context;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.internal.EntityInfoComponent;
+import org.terasology.entitySystem.event.Event;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.event.internal.EventReceiver;
+import org.terasology.entitySystem.event.internal.EventSystem;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+/**
+ * Test helper for listening to {@link Event}s
+ * <p>
+ * A test should instantiate a {@code TestEventReceiver}, execute some code, and then examine the list of entityRefs or
+ * events provided by {@link #getEntityRefs()} and {@link #getEvents()}.
+ * <p>
+ * Users can optionally supply a {@link BiConsumer} to handle the events with custom logic.
+ * <pre>
+ * {@literal
+ * TestEventReceiver receiver = new TestEventReceiver<>(context, DropItemEvent.class, (event, entity) -> {
+ *   // do something with the event or entity
+ * });
+ * }
+ * </pre>
+ *
+ * You can automatically unregister your receiver using a try-with-resources block ({@code
+ * TestEventReceiver} is {@link AutoCloseable}):
+ *
+ * <pre>
+ * {@literal
+ * try (TestEventReceiver<DropItemEvent> spy = new TestEventReceiver<>(getHostContext(), DropItemEvent.class)) {
+ *   drops = spy.getEntityRefs();
+ * }
+ * }
+ * </pre>
+ */
+public class TestEventReceiver<T extends Event> implements AutoCloseable, EventReceiver<T>{
+
+    private EventSystem eventSystem;
+    private Class<T> klass;
+    private List<EntityRef> entityRefs = new ArrayList<>();
+    private List<T> events = new ArrayList<>();
+    private BiConsumer<T, EntityRef> biConsumer = (a,b)-> { };
+
+    /**
+     * Constructs a new {@code TestEventReceiver} and registers it to listen for events.
+     *
+     * @param context the context object for the test; this should probably be obtained through
+     *                {@link ModuleTestingEnvironment#getHostContext()} and is needed so we can
+     *                obtain an {@link EventSystem} instance to register our event handler.
+     */
+    public TestEventReceiver(Context context, Class<T> klass, BiConsumer<T, EntityRef> biConsumer) {
+        this.biConsumer = biConsumer;
+        this.klass = klass;
+        eventSystem = context.get(EventSystem.class);
+        eventSystem.registerEventReceiver(this, klass, EntityInfoComponent.class);
+    }
+
+    public TestEventReceiver(Context context, Class<T> klass) {
+        this.klass = klass;
+        eventSystem = context.get(EventSystem.class);
+        eventSystem.registerEventReceiver(this, klass, EntityInfoComponent.class);
```

Comment:
I'm not familiar enough with the event system to know whether an entity is always guaranteed to have an `EntityInfoComponent`.  Even assuming that's the case, is there any reason to require it here?

Reply:
Actually I had to specify a component to work around an engine bug (wouldn't unregister "general" event receivers, only "specific" receives that specify a component). I'm not sure that all entities have an Info either. Also agree it should be extended to allow specifying component types

Reply:
Gotcha, thanks for the explanation.  If there's not already an issue for this in the engine (I didn't turn up anything, but I'm not confident enough in my search fu to be sure), you might consider submitting one.